packages/worker/src/worker.ts

```diff
@@ -233,81 +246,79 @@ class BaseWorker {
   }
 
   /**
-   * Do not make new poll requests.
+   * Do not make new poll requests, current poll request is not cancelled and may complete.
    */
   public suspendPolling(): void {
     if (this.state !== 'RUNNING') {
       throw new Error('Not running');
     }
-    this.nativeWorker.suspendPolling();
+    this.state = 'SUSPENDED';
   }
 
   /**
    * Allow new poll requests.
    */
   public resumePolling(): void {
-    if (this.state !== 'RUNNING') {
-      throw new Error('Not running');
+    if (this.state !== 'SUSPENDED') {
+      throw new Error('Not suspended');
     }
-    this.nativeWorker.resumePolling();
+    this.state = 'RUNNING';
   }
 
   public isSuspended(): boolean {
-    if (this.state !== 'RUNNING') {
-      throw new Error('Not running');
-    }
-    return this.nativeWorker.isSuspended();
+    return this.state === 'SUSPENDED';
   }
 
   shutdown(): void {
-    if (this.state !== 'RUNNING') {
-      throw new Error('Not running');
+    if (this.state !== 'RUNNING' && this.state !== 'SUSPENDED') {
+      throw new Error('Not running and not suspended');
     }
     this.state = 'STOPPING';
     this.nativeWorker.shutdown();
   }
 
   protected poller$(queueName: string): Observable<coresdk.Task> {
-    if (this.state !== 'INITIALIZED') {
-      throw new Error('Poller was aleady started');
-    }
-    return new Observable<coresdk.Task>((subscriber) => {
-      const startShutdownSequence = async (): Promise<void> => {
-        deregisterSignalHandlers();
-        this.shutdown();
-        await sleep(this.options.shutdownGraceTimeMs);
-        if (!subscriber.closed) {
-          subscriber.error(new Error('Timed out waiting while waiting for worker to shutdown gracefully'));
-        }
-      };
-      const deregisterSignalHandlers = () => {
-        for (const signal of this.options.shutdownSignals) {
-          process.off(signal, startShutdownSequence);
-        }
-      };
-      for (const signal of this.options.shutdownSignals) {
-        process.once(signal, startShutdownSequence);
-      }
-
-      this.state = 'RUNNING';
-
-      this.nativeWorker.poll(queueName, (err, buffer) => {
-        if (err && err.message.includes('[Core::shutdown]')) {
-          subscriber.complete();
-          this.activityHeartbeatSubject.complete();
-        } else if (buffer === undefined) {
-          subscriber.error(err);
-        } else {
-          const task = coresdk.Task.decode(new Uint8Array(buffer));
-          subscriber.next(task);
-        }
-      });
-
-      return function unsubscribe() {
-        // NOTE: We don't expose this observable directly so we don't have to shutdown here
-        deregisterSignalHandlers();
-      };
-    });
+    return merge(
+      this.stateSubject.pipe(
```

Comment:
I find this whole big chain of RxJS operators a bit difficult to follow... but I'm really not sure I have any suggestions on how to make it a little more clear since I've not used it. 

Maybe factoring out some of the inner pieces into their own functions so the top level piece reads a bit more clearly?